src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -731,6 +807,9 @@ public sealed override void WriteEntityReferenceLink(ODataEntityReferenceLink en
         /// </remarks>
         public sealed override Task WriteEntityReferenceLinkAsync(ODataEntityReferenceLink entityReferenceLink)
         {
+#if NETSTANDARD2_0 || NETCOREAPP3_1_OR_GREATER
+            OpenTelemetryHelper.ODataTemeletrySource.StartActivity(OpenTelemetryConstants.WriteEntityReferenceLink);
+#endif

```

Comment:
nitpick:

I think it may beneficial to create a helper method like `TelemetryHelper.StartActivity` and then inside that method we have the `#if NETCOREAPP3_1_OR_GREATER`. But this might be problematic if we want to use the returned activity object.